giraffe/src/utils/legend/staticLegend.ts

```diff
@@ -49,9 +60,42 @@ export const convertLineSpec = (
     values,
   }
 
-  const legendColumns = columnKeys.map(key => {
-    // TODO: should be using formatter from config
-    const column: string[] = mappings.map(fillColumn => fillColumn[key])
+  const additionalColumns = []
+  if (position === 'stacked') {
+    const stackedDomainValues = spec.stackedDomainValueColumn ?? []
+    additionalColumns.push({
+      key: valueColumnKey,
+      name: STACKED_LINE_CUMULATIVE,
+      type: spec.table.getColumnType(valueColumnKey),
+      colors,
+      values: sortOrder.map(index =>
+        valueFormatter(stackedDomainValues[index])
+      ),
+    })
+
```

Comment:
Stacked line graphs are line graphs that progressively add the previous line to the current line to determine the value. The line at the very top of the graph would be the sum of all the lines including itself.

With the above knowledge, the code is actually quite simple. `additionalColumns` are the extra "columns" (like "Latest _value") that stacked line graphs must have. What are they?

- `cumulative` the total value of the line plus all lines below it. each line has its own `_value` which is why a `cumulative` is also required
- `lines` is the count of how many lines are included for this line's `cumulative`. The line with the highest `lines` is the sum of all the other lines and itself. This column is especially important when negative values are present, because the highest line on the graph is not necessarily the line with the most `lines`.


See the Storybook examples to get a feel for what these columns mean.

Reply:
Regarding denseness of the code, that is caused by having a lot of necessary options that must be passed to a function. But rest assured, there isn't anything tricky in the code. `Prettier` is very liberal with spacing. Once you get used to reading the code looking at the indentation, there really isn't much going on in this block.

- create a variable
- make a column by pushing to `addtionalColumns`
- create a variable
- update a sort order
- make another column by pushing to `additionalColumns`